kernel/include/fs.h

```diff
@@ -54,6 +54,7 @@ typedef int (*readlink_type_t) (struct fs_node *, char * buf, size_t size);
 
 typedef struct fs_node {
 	char name[256];         /* The filename. */
+	char * path;            /* The path the node was opened at. */
```

Comment:
We probably want this at the end to maintain compatibility with any existing modules. (Not really a concern these days I suppose, but just in case...)